File: WowUp.Common/Models/Warcraft/Product.cs
using ProtoBuf;

namespace WowUp.Common.Models.Warcraft
{
    [ProtoContract]
    public class Product
    {
        [ProtoMember(1)]
        public string Name { get; set; }

        [ProtoMember(2)]
        public string Alias { get; set; }

        [ProtoMember(3)]
        public Client Client { get; set; }

        [ProtoMember(6)]
        public string Family { get; set; }
    }
}

File: wowup-electron/src/app/models/warcraft/product.ts
import { Field, Message, Type } from "protobufjs";
import { Client } from "./client";

@Type.d("Product")
export class Product extends Message<Product> {
  @Field.d(1, "string")
  public name: string;

  @Field.d(2, "string")
  public alias: string;

  @Field.d(3, Client)
  public client: Client;

  @Field.d(6, "string")
  public family: string;
}
